--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/aicwf_compat_8800d80.c
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/aicwf_compat_8800d80.c
@@ -40,7 +40,7 @@ int	rwnx_plat_userconfig_load_8800d80(st
     char *filename = FW_USERCONFIG_NAME_8800D80;
 
 #ifndef ANDROID_PLATFORM
-            sprintf(aic_fw_path, "%s/%s", aic_fw_path, "aic8800D80");
+            strlcat(aic_fw_path, "/aic8800D80", sizeof(aic_fw_path));
 #endif
 
     AICWFDBG(LOGINFO, "userconfig file path:%s \r\n", filename);
--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/aicwf_compat_8800d80x2.c
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/aicwf_compat_8800d80x2.c
@@ -41,7 +41,7 @@ int	rwnx_plat_userconfig_load_8800d80x2(
     char *filename = FW_USERCONFIG_NAME_8800D80X2;
 
 #ifndef ANDROID_PLATFORM
-    sprintf(aic_fw_path, "%s/%s", aic_fw_path, "aic8800D80X2");
+    strlcat(aic_fw_path, "/aic8800D80X2", sizeof(aic_fw_path));
 #endif
 
     AICWFDBG(LOGINFO, "userconfig file path:%s \r\n", filename);
--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_platform.c
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_platform.c
@@ -1669,7 +1669,7 @@ static int rwnx_plat_patch_load(struct r
     if(rwnx_hw->usbdev->chipid == PRODUCT_ID_AIC8800DC ||
         rwnx_hw->usbdev->chipid == PRODUCT_ID_AIC8800DW){
 #ifndef ANDROID_PLATFORM
-        sprintf(aic_fw_path, "%s/%s", aic_fw_path, "aic8800DC");
+        strlcat(aic_fw_path, "/aic8800DC", sizeof(aic_fw_path));
 #endif
         AICWFDBG(LOGINFO, "testmode=%d\n", testmode);
         if (chip_sub_id == 0) {
@@ -1808,7 +1808,7 @@ static int rwnx_plat_patch_load(struct r
         }
     } else if(rwnx_hw->usbdev->chipid == PRODUCT_ID_AIC8800D80N) {
 #ifndef ANDROID_PLATFORM
-        sprintf(aic_fw_path, "%s/%s", aic_fw_path, "aic8800D80N");
+        strlcat(aic_fw_path, "/aic8800D80N", sizeof(aic_fw_path));
 #endif
         if (testmode == FW_NORMAL_MODE) {
             ret = aicwf_plat_patch_load_8800d80n(rwnx_hw);
@@ -1838,7 +1838,7 @@ static int rwnx_plat_patch_load(struct r
     }
     else if(rwnx_hw->usbdev->chipid == PRODUCT_ID_AIC8800DLN) {
 #ifndef ANDROID_PLATFORM
-        sprintf(aic_fw_path, "%s/%s", aic_fw_path, "aic8800DLN");
+        strlcat(aic_fw_path, "/aic8800DLN", sizeof(aic_fw_path));
 #endif
         if (testmode == FW_NORMAL_MODE) {
             aicwf_patch_config_8800dln(rwnx_hw);
