--- a/arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts
@@ -229,6 +229,15 @@
 	status = "okay";
 };
 
+&gpio2 {
+	router_hat_pcie_enable_hog: router-hat-pcie-enable-hog {
+		gpio-hog;
+		gpios = <RK_PB6 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "router-hat-pcie-enable";
+	};
+};
+
 &gpu {
 	mali-supply = <&vdd_gpu_s0>;
 	status = "okay";
